cmd/utils/flags.go

```diff
@@ -757,14 +762,23 @@ func SetupNetwork(ctx *cli.Context) {
 		NetworkIdFlag.Value = 0
 		core.BlockReward = big.NewInt(1.5e+18)
 		core.ExpDiffPeriod = big.NewInt(math.MaxInt64)
+	default:
+		params.TargetGasLimit = common.String2Big(ctx.GlobalString(TargetGasLimitFlag.Name))
 	}
 }
 
-// SetupVM configured the VM package's global settings
-func SetupVM(ctx *cli.Context) {
-	vm.EnableJit = ctx.GlobalBool(VMEnableJitFlag.Name)
-	vm.ForceJit = ctx.GlobalBool(VMForceJitFlag.Name)
-	vm.SetJITCacheSize(ctx.GlobalInt(VMJitCacheFlag.Name))
+// MakeChainConfig reads the chain configuration from the given database or returns
+// an error if it failed.
+func MakeChainConfig(db ethdb.Database, ctx *cli.Context) (*core.ChainConfig, error) {
```

Comment:
None of the `MakeXYZ` methods return errors. This was done this way so that we can initialize configs and chains and whatnot in shorthand forms without having to deal with possible (and unhandleable!) errors.

As any error that the method itself cannot handle will anyway result in termination, the solution taken was the one with `MustMakeDataDir` (i.e. prefix with `Must`): if the method cannot handle it, it will `Fatalf` out. This was you can use `MakeXYZ` methods throughout this package without having to propagate errors all over the place that will jsut Fatalf at the very top of the stack.